src/index.js

```diff
@@ -272,6 +272,22 @@ const jsPrettier = (tree, {rules: {indent, eol}, jsBeautifyOptions}) => {
   return prettier(tree);
 };
 
+const addLang = (tree, {rules: {lang}}) => {
+  if (!lang) {
+    return tree;
+  }
+
+  tree.map(node => {
```

Comment:
Perhaps instead of the recursion suggested above, you should try the regular method [`walk`](https://github.com/posthtml/posthtml/blob/master/docs/api.md#treewalkcb--function)

Reply:
@Scrum tests failed because `tree.walk` seems undefined. 

![image](https://user-images.githubusercontent.com/26347874/83023706-e4827380-a04a-11ea-94ea-697dd9b9e528.png)

Reply:
Yes, because the tree has been modified in `clean`

https://github.com/posthtml/posthtml-beautify/blob/71aa6c2a70518d069f377f4880efb53ff67beb3e/src/index.js#L43-L73

and methods must be explicitly recreated or called with context

```js
const {walk} = require('posthtml/lib/api');
...
tree.walk = tree.walk || match;
```

or

```js
const {walk} = require('posthtml/lib/api');
...
walk.call(tree, ...)
```

Reply:
or can I put this  the `addLang` function before `clean` ?